File: backend/app/schemas/lead.py
from datetime import datetime
from uuid import UUID
from typing import Any
from pydantic import BaseModel


class LeadCreate(BaseModel):
    business_name: str
    industry: str | None = None
    description: str | None = None
    phone: str | None = None
    email: str | None = None
    website: str | None = None
    address: str | None = None
    city: str | None = None
    province: str = "Québec"
    postal_code: str | None = None
    latitude: float | None = None
    longitude: float | None = None
    google_place_id: str | None = None
    google_rating: float | None = None
    google_reviews: int | None = None
    yelp_url: str | None = None
    linkedin_url: str | None = None
    instagram_handle: str | None = None
    facebook_url: str | None = None
    owner_name: str | None = None
    owner_title: str | None = None
    owner_linkedin: str | None = None
    detected_language: str = "fr"
    source_flags: list[str] = []
    business_hours: dict[str, Any] | None = None


class LeadUpdate(BaseModel):
    status: str | None = None
    assigned_to: UUID | None = None
    next_followup_at: datetime | None = None
    ai_score: int | None = None
    score_rationale: str | None = None
    owner_name: str | None = None
    owner_title: str | None = None
    email: str | None = None
    phone: str | None = None


class LeadResponse(LeadCreate):
    id: UUID
    status: str
    ai_score: int | None
    score_rationale: str | None
    assigned_to: UUID | None
    last_contacted_at: datetime | None
    next_followup_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class LeadGeoJSON(BaseModel):
    id: UUID
    business_name: str
    status: str
    ai_score: int | None
    latitude: float
    longitude: float
    city: str | None
    industry: str | None

File: frontend/src/types/lead.ts
export type LeadStatus = "nouveau" | "contacté" | "réponse" | "rdv" | "fermé" | "perdu";

export interface Lead {
  id: string;
  business_name: string;
  industry: string | null;
  description: string | null;
  phone: string | null;
  email: string | null;
  website: string | null;
  address: string | null;
  city: string | null;
  province: string;
  postal_code: string | null;
  latitude: number | null;
  longitude: number | null;
  google_place_id: string | null;
  google_rating: number | null;
  google_reviews: number | null;
  yelp_url: string | null;
  linkedin_url: string | null;
  instagram_handle: string | null;
  facebook_url: string | null;
  owner_name: string | null;
  owner_title: string | null;
  owner_linkedin: string | null;
  detected_language: string;
  status: LeadStatus;
  ai_score: number | null;
  score_rationale: string | null;
  source_flags: string[];
  assigned_to: string | null;
  last_contacted_at: string | null;
  next_followup_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface Interaction {
  id: string;
  type: string;
  notes: string | null;
  metadata: Record<string, unknown> | null;
  occurred_at: string;
}

export const STATUS_LABELS: Record<LeadStatus, string> = {
  nouveau: "Nouveau",
  contacté: "Contacté",
  réponse: "Réponse",
  rdv: "RDV",
  fermé: "Fermé",
  perdu: "Perdu",
};

export const STATUS_COLORS: Record<LeadStatus, string> = {
  nouveau: "bg-blue-100 text-blue-700 border-blue-200",
  contacté: "bg-amber-100 text-amber-700 border-amber-200",
  réponse: "bg-emerald-100 text-emerald-700 border-emerald-200",
  rdv: "bg-violet-100 text-violet-700 border-violet-200",
  fermé: "bg-gray-100 text-gray-600 border-gray-200",
  perdu: "bg-red-100 text-red-600 border-red-200",
};

export const STATUS_DOT: Record<LeadStatus, string> = {
  nouveau: "bg-blue-500",
  contacté: "bg-amber-500",
  réponse: "bg-emerald-500",
  rdv: "bg-violet-500",
  fermé: "bg-gray-400",
  perdu: "bg-red-500",
};

export const PIPELINE_COLUMNS: LeadStatus[] = [
  "nouveau",
  "contacté",
  "réponse",
  "rdv",
  "fermé",
  "perdu",
];

export const INDUSTRY_LABELS: Record<string, string> = {
  restaurant: "Restaurant",
  cafe: "Café",
  hotel: "Hôtel",
  plombier: "Plombier",
  electricien: "Électricien",
  serrurier: "Serrurier",
  peintre: "Peintre",
  menuisier: "Menuisier",
  autre: "Autre",
};

export const SOURCE_ICONS: Record<string, string> = {
  google_maps: "🗺️",
  pages_jaunes: "📒",
  yelp: "⭐",
  linkedin: "💼",
  instagram: "📸",
  facebook: "👥",
};
